src/custom/utils/trade.ts

```diff
@@ -88,12 +89,34 @@ export async function postOrder(params: PostOrderParams): Promise<string> {
     partiallyFillable: false // Always fill or kill
   }
 
-  const signature = (await signOrder({
+  let signature: Signature | null = null
+  let signingScheme = DEFAULT_SIGNING_SCHEME
+
+  const signatureParams: SignOrderParams = {
     chainId,
     signer,
     order: unsignedOrder,
-    signingScheme: SIGNING_SCHEME
-  })) as EcdsaSignature // Only ECDSA signing supported for now
+    signingScheme
+  }
+
+  try {
+    signature = (await signOrder(signatureParams)) as EcdsaSignature // Only ECDSA signing supported for now
+  } catch (e) {
+    if (e.code === METAMASK_SIGNATURE_ERROR) {
+      // We tried to sign order the nice way.
+      // That works fine for regular MM addresses. Does not work for Hardware wallets, though.
+      // See https://github.com/MetaMask/metamask-extension/issues/10240#issuecomment-812672324
```

Comment:
I see the issue is relevant. Although I'm not sure about the comment you point out there. If i understand the comment is actually very bad advice:
![image](https://user-images.githubusercontent.com/2352112/114578940-ac454d00-9c7d-11eb-818a-bf3e335ae716.png)

suggests to import the seed from the ledger in metamaks 🙈

Reply:
Oops, copied the wrong comment.
Meant to copy the one that talks about this being worked on their next sprint.
Will update.